packages/design/src/utils/match/matchers.ts

```diff
@@ -0,0 +1,20 @@
+import { displayDate, displayDateTime } from 'shared/services/loc';
+import { MatchCallback } from './match';
+
+export const dateMatcher =
+  <T>(datePropNames: (keyof T & string)[]): MatchCallback<T> =>
+  (targetValue, searchValue, propName) => {
+    if (datePropNames.includes(propName)) {
+      return displayDate(targetValue).toLocaleLowerCase().includes(searchValue);
+    }
+  };
```

Comment:
my brain cramped reading this lol, i would save shortcut return statements for when its short and easy to read. I also tend to favor using keyword `function` whenever i can b/c it clearly states it's a function and when it concerns typing, it's easier to reason:

```suggestion
export function dateMatcher<T>(
  datePropNames: (keyof T & string)[]
): MatchCallback<T> {
  return (targetValue, searchValue, propName) => {
    if (datePropNames.includes(propName)) {
      return displayDate(targetValue).toLocaleLowerCase().includes(searchValue);
    }
  };
}
```